src/core/shards/wires.cpp

```diff
@@ -631,6 +633,28 @@ struct Recur : public WireBase {
 };
 
 struct BaseRunner : public WireBase {
+  std::deque<ParamVar> _vars;
+
+  SHTypeInfo compose(const SHInstanceData &data) {
+    auto res = WireBase::compose(data);
+
+    if (capturing) {
+      // build the list of variables to capture and inject into spawned chain
+      _vars.clear();
+      for (auto &require : wireValidation.requiredInfo) {
+        if (!require.global) {
+          SHVar ctxVar{};
+          ctxVar.valueType = ContextVar;
+          ctxVar.payload.stringValue = require.name;
+          auto &p = _vars.emplace_back();
```

Comment:
Can you change it to `_vars.emplace_back(ctxVar);`?

Reply:
No, it's on purpose to trigger `operator=` clone, I think.

Reply:
Constructor with SHVar& has clone too

Reply:
did you test it? hmm can swap to vector if so

Reply:
I didn't but it should work on both I think. 
If it doesn't work because the `TParamVar(SHVar initialValue)` is explicit, just keep this.